TypeHandler 추가 및 xml 파일 추가

dlrtn opened the pull request:
추가 후 테스트 완료하였습니다.
TypeHandler 부분 정리 부족하여 코드 구현이 부족할 수 있습니다
달게받겟습니다.. 컴온..

dlrtn committed:
회원가입 기능 추가 및 의존성 추가

회원가입 기능 추가하였으나 404 에러 발생합니다. 버그 찾는 중이나 찾지 못했습니다. 내일 오전 중으로 디버깅 작업 예정입니다. 아래는 테스트/디버깅 진행 결과인데, 참고하시어 문제되는 부분 있으면 말씀해주시면 좋겠습니다.

dlrtn committed:
중간 커밋

Commit:
gradle 의존성 정리

Commit:
유저 코드 리펙토링

dlrtn committed:
Mybatis 테스트 작성

woojae9488 committed:
MyBatis 테스트 정리

dlrtn committed:
변수명 변경 및 유저 저장 로직? 테스트

dlrtn committed:
코드 보기좋게

dlrtn committed:
SignUpRequest 추가 및 table,  entity object mapping 수정

dlrtn committed:
SignUpRequest 추가 및 table,  entity object mapping 수정

dlrtn committed:
LocalDateTimeTypeHandler, mybatis-config.xml 추가 및 테스트 완료

dlrtn committed:
Mapper unit Test 수정

woojae9488 commented on src/main/java/com/dlrtn/websocket/chat/config/mybatis/LocalDateTimeTypeHandler.java:
```diff
@@ -0,0 +1,48 @@
+import org.apache.ibatis.type.JdbcType;
+import org.apache.ibatis.type.MappedTypes;
+
+@MappedTypes(LocalDateTime.class)
+public class LocalDateTimeTypeHandler extends BaseTypeHandler<LocalDateTime> {
```
LocalDateTime을 TIMESTAMP 타입으로 바꾸는 타입 핸들러는 기존에 MyBatis에 등록이 되어 있어 별도 선언이 필요 없을 거 같습니다.
(https://mybatis.org/mybatis-3/ko/configuration.html#typeHandlers)

woojae9488 commented on src/main/resources/mybatis-config.xml:
```diff
@@ -0,0 +1,3 @@
+<typeHandlers>
+    <typeHandler handler="org.mybatis.example.LocalDateTimeTypeHandler"/>
+</typeHandlers>
```
타입핸들러는 bean config로도 설정이 가능해서 되도록이면 xml 파일로 관리하기 보다는 MybatisConfig 만들어서 빈 등록하는게 좋을 거 같습니다.

woojae9488 reviewed (changes requested):
기존 `feature/user_signup` 브랜치의 변경 사항들도 겹쳐서 보이는 거 같은데요.
기능 브랜치 분리가 제대로 안된 거 같습니다.
확인 부탁드릴게요

dlrtn committed:
Merge branch 'develop' into feature/user_signup

dlrtn committed:
레퍼런스
https://dev-gorany.tistory.com/212?category=901854

websocket chating 로직 추가, front 단에서 오류 나는 것 확인

11일에 확인 바람

dlrtn committed:
Merge remote-tracking branch 'origin/feature/user_signup' into feature/user_signup
